scripts/courseGenerator/CourseGeneratorInterface.lua

```diff
@@ -71,6 +72,36 @@ function CourseGeneratorInterface.generate(fieldPolygon,
 	-----------------------------------------------------------------------------------------------------------------------
 	local islandNodes = Island.findIslands(Polygon:new(CourseGenerator.pointsToXy(fieldPolygon)))
 
+	--------------------------------------------------------------------------------------------------------------------
+	-- Field Margin
+	-----------------------------------------------------------------------------------------------------------------------
+	poly = Polygon:new(CourseGenerator.pointsToXy(fieldPolygon))
+	local centerX, centerZ = poly:getCenter()
+	centerZ = centerZ * -1
+	
+	local fieldPolygonWithMargin = {}
+
+	if fieldMargin > 0 then
+		for i, point in ipairs(fieldPolygon) do
+			local p = shallowCopy(point)
+			if p.x > centerX then

```

Comment:
This works fine for rectangular and probably most convex polygons only.

Reply:
Thanks for your feedback! I reviewed the calculateHeadlandtrack function, but I'm not certain on the best way to incorporate this into it. Any help you could provide would be greatly appreciated!

Reply:
No problem, I created a field-margin branch, please create a new pull request to that branch (instead of main) and we go from there.